clients/cobol-lsp-vscode-extension/src/__tests__/JavaCheckTest.ts

```diff
@@ -27,8 +28,93 @@ describe("Check Java version", () => {
         expect(JavaCheck.isJavaVersionSupported('openjdk version "12.0.1" 2018-10-16')).toBeTruthy();
         expect(JavaCheck.isJavaVersionSupported("java 11 2018-09-25")).toBeTruthy();
         expect(JavaCheck.isJavaVersionSupported("java 12 2019-03-19")).toBeTruthy();
+        expect(JavaCheck.isJavaVersionSupported("java 13 2019-09-17")).toBeTruthy();
+        expect(JavaCheck.isJavaVersionSupported("java 14 2020-03-17")).toBeTruthy();
     });
-    it("checks if Java version is not supported", async () => {
+
+    it("when version is not supported", async () => {
         expect(JavaCheck.isJavaVersionSupported('java version "1.7.0_131"')).toBeFalsy();
+        expect(JavaCheck.isJavaVersionSupported('java version "1.5.0_22"')).toBeFalsy();
+    });
+});
+
+describe("Checks Java installation", () => {
+
+    let javaCheck: JavaCheck;
+    let map: any;
+    let checkFn: any;
+    let stderrFn: any;
+    const expectedErrMsgJavaVersionFound = "Java version 8 expected";
+    const expectedErrMsgJavaVersionNotFound = "Java 8 is not found";
+
+    beforeEach(() => {
+        javaCheck = new JavaCheck();
+        checkFn = (event, callbackFn) => {
+            map[event] = callbackFn;
+        };
+        stderrFn = (event, callbackFn) => {
+            map[event] = callbackFn;
+        };
+        map = {};
+    });
+
+    it("when required version is supported", async () => {
+        (cp as any).spawn = jest.fn().mockReturnValue({stderr: {on: stderrFn}, on: jest.fn()});
+        const promise = javaCheck.isJavaInstalled();
+        map.data("java 11 2018-09-25");
+
+        try {
+            expect(await promise).toEqual(undefined);
+        } catch (e) {
+            expect(e.toString()).toEqual(expectedErrMsgJavaVersionFound);
+        }
+    });
+
+    it("when required version is not supported", async () => {
+        (cp as any).spawn = jest.fn().mockReturnValue({stderr: {on: stderrFn}, on: jest.fn()});
+        const promise = javaCheck.isJavaInstalled();
+        map.data('java version "1.5.0_22"');
+
+        try {
+            await promise;
+        } catch (e) {
+            expect(e.toString()).toEqual(expectedErrMsgJavaVersionFound);
+        }
+    });
+
+    it("when 'error' event is emitted  - spawned", async () => {
+        (cp as any).spawn = jest.fn().mockReturnValue({stderr: {on: jest.fn()}, on: checkFn});
+        const promise = javaCheck.isJavaInstalled();
+        map.error("Error: spawn java ENOENT");
+
+        try {
+            await promise;
+        } catch (e) {
+            expect(e.toString()).toEqual(expectedErrMsgJavaVersionNotFound);
+        }
+    });
```

Comment:
Just a tiny suggestion: this try/catch for negative cases could be probably extracted as separate function

Reply:
Instead of try/catch block, you could use this:
`await expect(promise).rejects.toEqual(expectedErrMsgJavaVersionNotFound);`

Reply:
> Just a tiny suggestion: this try/catch for negative cases could be probably extracted as separate function

@zacanbrcom Thanks done

Reply:
> Instead of try/catch block, you could use this:
> `await expect(promise).rejects.toEqual(expectedErrMsgJavaVersionNotFound);`

@grianbrcom Thanks, altered